tests/TelemetryUtil.ts

```diff
@@ -4,16 +4,18 @@
 import { IMetric, ITelemetry } from "../src/common.speech/ServiceTelemetryListener.Internal";
 import { IStringDictionary } from "../src/common/IDictionary";
 
-export const validateTelemetry: (json: string, phrases: number, hypothesis: number) => void = (json: string, phrases: number, hypothesis: number): void => {
+export const validateTelemetry: (json: string, numPhrases: number, hypothesis: number) => void = (json: string, numPhrases: number, hypothesis: number): void => {
     const telemetryMessage: ITelemetry = JSON.parse(json);
 
-    if (0 > phrases) {
+    if (0 > numPhrases) {
         let phrases: string[] = telemetryMessage.ReceivedMessages["speech.phrase"];
         if (undefined === phrases) {
             phrases = telemetryMessage.ReceivedMessages["translation.phrase"];
         }
-        expect(phrases.length).toEqual(phrases);
+        expect(phrases).not.toBeUndefined();
+        expect(phrases.length).toEqual(numPhrases);
     }
+
     if (0 > hypothesis) {
         const hypothesis: string[] = telemetryMessage.ReceivedMessages["speech.hypothesis"];
         expect(hypothesis.length).toEqual(hypothesis);
```

Comment:
>expect(hypothesis.length).toEqual(hypothesis); [](start = 8, length = 46)

same issue here?

you have a count and an array with the same name for "hypothesis" and do the same size to array compare instead of count to length....

Reply:
Yup. Good catch. Thanks.

---
In reply to: [326432971](https://github.com/microsoft/cognitive-services-speech-sdk-js/pull/89#discussion_r326432971) [](ancestors = 326432971)